File: Server_Root/Server/RoomServer/Messages/GameEventStopToC.cpp
#include "stdafx.h"
#include "GameEventStopToC.h"
#include "GameEventStopToC_Data_generated.h"

GameEventStopToC::GameEventStopToC() : m_Builder(1024)
{
}

GameEventStopToC::~GameEventStopToC()
{
}

unsigned short GameEventStopToC::GetID()
{
	return MESSAGE_ID;
}

IMessage* GameEventStopToC::Clone()
{
	return NULL;
}

const char* GameEventStopToC::Serialize(int* pLength)
{
	FBSData::Vector3 pos(m_vec3Pos.x, m_vec3Pos.y, m_vec3Pos.z);

	GameEventStopToC_DataBuilder data_builder(m_Builder);
	data_builder.add_PlayerIndex(m_nPlayerIndex);
	data_builder.add_EventTime(m_lEventTime);
	data_builder.add_Pos(&pos);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool GameEventStopToC::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<GameEventStopToC_Data>((const void*)pChar);

	m_nPlayerIndex = data->PlayerIndex();
	m_lEventTime = data->EventTime();
	m_vec3Pos.x = data->Pos()->x();
	m_vec3Pos.y = data->Pos()->y();
	m_vec3Pos.z = data->Pos()->z();

	return true;
}

File: Client_Root/Client/Assets/Scripts/Network/Messages/ToClient/GameEventStopToC.cs
using UnityEngine;
using FlatBuffers;

public class GameEventStopToC : IMessage
{
    public const ushort MESSAGE_ID = MessageID.GameEventStopToC_ID;

    public int m_nPlayerIndex;
    public long m_lEventTime;
    public Vector3 m_vec3Pos;

    public ushort GetID()
    {
        return MESSAGE_ID;
    }

    public IMessage Clone()
    {
        return null; 
    }

    public byte[] Serialize()
    {
        FlatBufferBuilder builder = new FlatBufferBuilder(1024);

        var pos = FBSData.Vector3.CreateVector3(builder, m_vec3Pos.x, m_vec3Pos.y, m_vec3Pos.z);

        GameEventStopToC_Data.StartGameEventStopToC_Data(builder);
        GameEventStopToC_Data.AddPlayerIndex(builder, m_nPlayerIndex);
        GameEventStopToC_Data.AddEventTime(builder, m_lEventTime);
        GameEventStopToC_Data.AddPos(builder, pos);
        var data = GameEventStopToC_Data.EndGameEventStopToC_Data(builder);

        builder.Finish(data.Value);

        return builder.SizedByteArray();
    }

    public bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = GameEventStopToC_Data.GetRootAsGameEventStopToC_Data(buf);

        m_nPlayerIndex = data.PlayerIndex;
        m_lEventTime = data.EventTime;
        m_vec3Pos = new Vector3(data.Pos.X, data.Pos.Y, data.Pos.Z);

        return true;
    }
}
